--- lms.orig/index.php	Tue Apr 15 00:53:15 2003
+++ lms/index.php	Tue Apr 15 00:52:47 2003
@@ -53,14 +53,14 @@
 									
 // Define directories and configuration vars
 
-$_SYSTEM_DIR = (! $_CONFIG[directories]['sys_dir'] ? getcwd() : $_CONFIG[directories]['sys_dir']);
-$_BACKUP_DIR = (! $_CONFIG[directories]['backup_dir'] ? $_SYSTEM_DIR."/backups/" : $_CONFIG[directories]['backup_dir']);
-$_LIB_DIR = (! $_CONFIG[directories]['lib_dir'] ? $_SYSTEM_DIR."/lib/" : $_CONFIG[directories]['lib_dir']);
-$_MODULES_DIR = (! $_CONFIG[directories]['modules_dir'] ? $_SYSTEM_DIR."/modules/" : $_CONFIG[directories]['modules_dir']);
-$_SMARTY_DIR = (! $_CONFIG[directories]['smarty_dir'] ? $_LIB_DIR."/Smarty/" : $_CONFIG[directories]['smarty_dir']);
-$_SMARTY_COMPILE_DIR = (! $_CONFIG[directories]['smarty_compile_dir'] ? $_SYSTEM_DIR."/templates_c" : $_CONFIG[directories]['smarty_compile_dir']);
-$_SMARTY_TEMPLATES_DIR = (! $_CONFIG[directories]['smarty_templates_dir'] ? $_SYSTEM_DIR."/templates" : $_CONFIG[directories]['smarty_templates_dir']);
-$_ADODB_DIR = (! $_CONFIG[directories]['adodb_dir'] ? $_LIB_DIR."/adodb/" : $_CONFIG[directories]['adodb_dir']);
+$_SYSTEM_DIR = (! $_CONFIG[directories]['sys_dir'] ? '/usr/lib/lms/' : $_CONFIG[directories]['sys_dir']);
+$_BACKUP_DIR = (! $_CONFIG[directories]['backup_dir'] ? '/var/lib/lms/backups/' : $_CONFIG[directories]['backup_dir']);
+$_LIB_DIR = (! $_CONFIG[directories]['lib_dir'] ? $_SYSTEM_DIR.'/lib/' : $_CONFIG[directories]['lib_dir']);
+$_MODULES_DIR = (! $_CONFIG[directories]['modules_dir'] ? $_SYSTEM_DIR.'/modules/' : $_CONFIG[directories]['modules_dir']);
+$_SMARTY_DIR = (! $_CONFIG[directories]['smarty_dir'] ? '/usr/share/pear/Smarty/' : $_CONFIG[directories]['smarty_dir']);
+$_SMARTY_COMPILE_DIR = (! $_CONFIG[directories]['smarty_compile_dir'] ? '/var/lib/lms/templates_c/' : $_CONFIG[directories]['smarty_compile_dir']);
+$_SMARTY_TEMPLATES_DIR = (! $_CONFIG[directories]['smarty_templates_dir'] ? $_SYSTEM_DIR.'/templates/' : $_CONFIG[directories]['smarty_templates_dir']);
+$_ADODB_DIR = (! $_CONFIG[directories]['adodb_dir'] ? '/usr/share/php/adodb/' : $_CONFIG[directories]['adodb_dir']);
 $_TIMEOUT = (! $_CONFIG[phpui]['timeout'] ? 600 : $_CONFIG[phpui]['timeout']);
 $_FORCE_SSL = chkconfig($_CONFIG[phpui]['force_ssl']);
 
