--- lms/sample/lms.ini.orig	Mon Feb 24 20:01:11 2003
+++ lms/sample/lms.ini	Mon Feb 24 19:59:54 2003
@@ -150,7 +150,7 @@
 ; lib_dir.
 ; http://smarty.php.net
 
-; smarty_dir		= /usr/lib/php/Smarty
+smarty_dir		= /usr/share/pear/Smarty
 
 ; Katalog kompilacji Smartyego. Miejsce gdzie Smarty psuje nasze
 ; templejty. Domy¶lnie to templates_c w katalogu sysdir.
@@ -166,7 +166,7 @@
 ; podkatalog adodb w lib_dir.
 ; http://php.weblogs.com/ADOdb
 ;
-; adodb_dir		= /usr/lib/php/adodb
+adodb_dir		= /usr/share/pear/adodb
 
 ; ---------------------------------------------------------------------------
 ; Notify configuration.
