diff -urP lms.orig/bin/lms-mgc lms/bin/lms-mgc
--- lms.orig/bin/lms-mgc	Thu Mar 27 05:13:30 2003
+++ lms/bin/lms-mgc	Thu Mar 27 05:06:31 2003
@@ -389,7 +389,7 @@
 		# BUILD NODELIST
 		#
 		#
-		$dbq = $dbase->prepare("SELECT ipaddr, mac, access, id, ownerid, name FROM nodes");
+		$dbq = $dbase->prepare("SELECT ipaddr, mac, access, id, ownerid, name, filename FROM nodes");
 		$dbq->execute();
 		my %maclist;
 		my %accesslist;
@@ -397,6 +397,7 @@
 		my %namelist;
 		my %ucnamelist;
 		my %ownerlist;
+		my %filenamelist;
 		
 		while (my $row = $dbq->fetchrow_hashref())
 		{
@@ -406,6 +407,7 @@
 			$namelist{$row->{'ipaddr'}} = lc($row->{'name'});
 			$ucnamelist{$row->{'ipaddr'}} = uc($row->{'name'});
 			$ownerlist{$row->{'ipaddr'}} = $row->{'ownerid'};
+			$filenamelist{$row->{'ipaddr'}} = $row->{'filename'};
 		}
 				
 		
@@ -514,6 +516,7 @@
 			my $NAME = $ucnamelist{$ipaddr} || '';
 			my $name = $namelist{$ipaddr} || '';
 			my $mac = $maclist{$ipaddr} || '';
+			my $fname = $filenamelist{$ipaddr} || '';
 			my $macs = lc($mac);
 			$macs =~ s/://g;
 			my $owner = $ownerlist{$ipaddr} || '';
@@ -538,6 +541,7 @@
 			$rule =~ s/\%name/$name/g;
 			$rule =~ s/\%SMAC/$macs/g;
 			$rule =~ s/\%MAC/$mac/g;
+			$rule =~ s/\%filename/$fname/g;
 			$rule =~ s/\%OWNER/$owner/g;
 			$rule =~ s/\%TIMES/$times/g;
 			$rule =~ s/\%TIME/$time/g;
diff -urP lms.orig/doc/lms.mysql lms/doc/lms.mysql
--- lms.orig/doc/lms.mysql	Thu Mar 27 05:13:30 2003
+++ lms/doc/lms.mysql	Thu Mar 27 05:06:31 2003
@@ -94,6 +94,7 @@
   `creatorid` int(11) NOT NULL default '0',
   `modid` int(11) NOT NULL default '0',
   `access` char(1) NOT NULL default 'Y',
+  `filename` varchar(255) NOT NULL default '',    
   PRIMARY KEY  (`id`),
   UNIQUE KEY `nodeid_2` (`id`,`name`),
   KEY `nodeid` (`id`)
diff -urP lms.orig/lib/LMS.class.php lms/lib/LMS.class.php
--- lms.orig/lib/LMS.class.php	Thu Mar 27 05:13:30 2003
+++ lms/lib/LMS.class.php	Thu Mar 27 05:12:50 2003
@@ -553,7 +553,7 @@
 
 	function GetNode($id)
 	{
-		if($return = $this->ADB->GetRow("SELECT id, name, ownerid, ipaddr, mac, access, creationdate, moddate, creatorid, modid FROM nodes WHERE id=?",array($id)))
+		if($return = $this->ADB->GetRow("SELECT id, name, ownerid, ipaddr, mac, access, creationdate, moddate, creatorid, modid, filename FROM nodes WHERE id=?",array($id)))
 		{
 			$return[createdby] = $this->GetAdminName($return[creatorid]);
 			$return[modifiedby] = $this->GetAdminName($return[modid]);
@@ -1341,7 +1341,7 @@
 	{
 		$this->SetTS("nodes");
 
-		if($this->ADB->Execute("INSERT INTO nodes (name, mac, ipaddr, ownerid, creatorid, creationdate) VALUES (?, ?, ?, ?, ?, ".$this->sqlTSfmt().")",array(strtoupper($nodedata[name]),strtoupper($nodedata[mac]),$nodedata[ipaddr],$nodedata[ownerid],$this->SESSION->id)))
+		if($this->ADB->Execute("INSERT INTO nodes (name, mac, ipaddr, ownerid, creatorid, creationdate, filename) VALUES (?, ?, ?, ?, ?, ".$this->sqlTSfmt().", ?)",array(strtoupper($nodedata[name]),strtoupper($nodedata[mac]),$nodedata[ipaddr],$nodedata[ownerid],$this->SESSION->id,$nodedata[filename])))
 			return $this->ADB->GetOne("SELECT MAX(id) FROM nodes");
 		else
 			return FALSE;
@@ -1513,7 +1513,7 @@
 	function NodeUpdate($nodedata)
 	{
 		$this->SetTS("nodes");
-		return $this->ADB->Execute("UPDATE nodes SET name=?, ipaddr=?, mac=?, moddate=".$this->sqlTSfmt().", modid=?, access=?, ownerid=? WHERE id=?",array(strtoupper($nodedata[name]), $nodedata[ipaddr], strtoupper($nodedata[mac]), $this->SESSION->id, $nodedata[access], $nodedata[ownerid], $nodedata[id]));
+		return $this->ADB->Execute("UPDATE nodes SET name=?, ipaddr=?, mac=?, moddate=".$this->sqlTSfmt().", modid=?, access=?, ownerid=?, filename=? WHERE id=?",array(strtoupper($nodedata[name]), $nodedata[ipaddr], strtoupper($nodedata[mac]), $this->SESSION->id, $nodedata[access], $nodedata[ownerid], $nodedata[filename], $nodedata[id]));
 	}
 
 	function GetUsersWithTariff($id)
diff -urP lms.orig/modules/nodeedit.php lms/modules/nodeedit.php
--- lms.orig/modules/nodeedit.php	Thu Mar 27 05:13:30 2003
+++ lms/modules/nodeedit.php	Thu Mar 27 05:06:31 2003
@@ -108,6 +108,7 @@
 	$nodeinfo[ipaddr] = $nodeedit[ipaddr];
 	$nodeinfo[access] = $nodeedit[access];
 	$nodeinfo[ownerid] = $nodeedit[ownerid];
+	$nodeinfo[filename] = $nodeedit[filename];
 
 	if(!$error)
 	{
diff -urP lms.orig/templates/nodeaddbox.html lms/templates/nodeaddbox.html
--- lms.orig/templates/nodeaddbox.html	Thu Mar 27 05:13:30 2003
+++ lms/templates/nodeaddbox.html	Thu Mar 27 05:07:51 2003
@@ -30,6 +30,14 @@
 	</TR>
 	<TR CLASS="LIGHT">
 		<TD WIDTH="1%" CLASS="FLEFT">
+			<IMG SRC="img/save.gif" BORDER="0" ALT="Boot image:">
+		</TD>
+		<TD WIDTH="99%" CLASS="FRIGHT" COLSPAN="2">
+			<INPUT TYPE="TEXT" NAME="nodedata[filename]" VALUE="{$nodedata.filename}">
+		</TD>
+	</TR>
+	<TR CLASS="LIGHT">
+		<TD WIDTH="1%" CLASS="FLEFT">
 			<IMG SRC="img/n{if $nodedata.access neq "Y" && $nodedata.access neq ""}no{/if}access.gif" BORDER="0" ALT="">
 		</TD>
 		<TD WIDTH="99%" CLASS="FRIGHT" COLSPAN="2">
diff -urP lms.orig/templates/nodeeditbox.html lms/templates/nodeeditbox.html
--- lms.orig/templates/nodeeditbox.html	Thu Mar 27 05:13:30 2003
+++ lms/templates/nodeeditbox.html	Thu Mar 27 05:06:31 2003
@@ -31,6 +31,14 @@
 	</TR>
 	<TR CLASS="LIGHT">
 		<TD WIDTH="1%" CLASS="FLEFT">
+			<IMG SRC="img/save.gif" BORDER="0" ALT="Boot image:">
+		</TD>
+		<TD WIDTH="99%" CLASS="FRIGHT" COLSPAN="2">
+			<INPUT TYPE="TEXT" NAME="nodeedit[filename]" VALUE="{$nodeinfo.filename}" CLASS="BOLD">
+		</TD>
+	</TR>
+	<TR CLASS="LIGHT">
+		<TD WIDTH="1%" CLASS="FLEFT">
 			<IMG SRC="img/network.gif" BORDER="0" ALT="Sieæ:">
 		</TD>
 		<TD WIDTH="99%" CLASS="FRIGHT" COLSPAN="2">
diff -urP lms.orig/templates/nodeinfobox.html lms/templates/nodeinfobox.html
--- lms.orig/templates/nodeinfobox.html	Thu Mar 27 05:13:30 2003
+++ lms/templates/nodeinfobox.html	Thu Mar 27 05:06:31 2003
@@ -24,6 +24,16 @@
 			{$nodeinfo.mac}
 		</TD>
 	</TR>
+	{if $nodeinfo.filename neq ""}
+	<TR CLASS="LIGHT">
+		<TD WIDTH="1%" CLASS="FLEFT">
+			<IMG SRC="img/save.gif" BORDER="0" ALT="Boot image:">
+		</TD>
+		<TD WIDTH="99%" CLASS="FRIGHT" COLSPAN="2">
+			<B>{$nodeinfo.filename}</B>
+		</TD>
+	</TR>
+	{/if}
 	<TR CLASS="LIGHT">
 		<TD WIDTH="1%" CLASS="FLEFT">
 			<IMG SRC="img/network.gif" BORDER="0" ALT="Sieæ:">
