--- index.php	2012-04-25 01:23:18.000000000 +0300
+++ index.php	2012-04-25 16:19:21.101145466 +0300
@@ -184,11 +184,28 @@
 header('X-Powered-By: LMS/'.$layout['lmsv']);
 
 // Check privileges and execute modules
-if($AUTH->islogged)
-{
+if ($AUTH->islogged) {
+	// Load plugin files and register hook callbacks
+	$plugins = preg_split('/[;,\s\t\n]+/', $CONFIG['phpui']['plugins'], -1, PREG_SPLIT_NO_EMPTY);
+	if (!empty($plugins))
+		foreach ($plugins as $plugin_name)
+			require LIB_DIR . '/plugins/' . $plugin_name . '.php';
+
+	$res = $LMS->ExecHook('access_table_init', array('accesstable' => $access['table']));
+	if (isset($res['accesstable']))
+		$access['table'] = $res['accesstable'];
+
 	$module = isset($_GET['m']) ? preg_replace('/[^a-zA-Z0-9_-]/', '', $_GET['m']) : '';
 	$deny = $allow = FALSE;
 
+	$res = $LMS->ExecHook('module_load_before', array('module' => $module));
+	if ($res['abort']) {
+		$SESSION->close();
+		$DB->Destroy();
+		die;
+	}
+	$module = $res['module'];
+
 	if ($AUTH->passwdrequiredchange)
 		$module = 'chpasswd';
 
@@ -197,12 +214,6 @@
 		$module = $CONFIG['phpui']['default_module'];
 	}
 
-	// Load plugin files and register hook callbacks
-	$plugins = preg_split('/[;,\s\t\n]+/', $CONFIG['phpui']['plugins'], -1, PREG_SPLIT_NO_EMPTY);
-	if (!empty($plugins))
-		foreach ($plugins as $plugin_name)
-			require LIB_DIR . '/plugins/' . $plugin_name . '.php';
-
 	if (file_exists(MODULES_DIR.'/'.$module.'.php'))
 	{
 		$allow = !$AUTH->id || (!empty($access['allow']) && preg_match('/'.$access['allow'].'/i', $module));
