--- a/userpanel/modules/finances/invoice_pdf.php	2015-12-07 18:18:16.595346482 +0100
+++ b/userpanel/modules/finances/invoice_pdf.php	2015-12-07 19:21:48.574802326 +0100
@@ -94,8 +97,16 @@
 			array($msgid, $invoice['customerid'], '', MSG_SENT, NULL));
 		$DB->CommitTrans();
 	}
-	
-	close_pdf($pdf);
+
+	if (isset($docnumber)) {
+		$filename = ConfigHelper::getConfig('invoices.file_name', 'file.pdf');
+		$filename = str_replace('%number', $docnumber, $filename);
+		$filename = preg_replace('/[\/\\ ]/', '_', $filename);
+	} else
+		$filename = null;
+
+	close_pdf($pdf, $filename);
+
 	die;
 }
 
