diff --git a/img/common.js b/img/common.js
index a7aa95b..173885b 100644
--- a/img/common.js
+++ b/img/common.js
@@ -87,10 +87,10 @@ function openSelectWindow2(theURL, winName, myWidth, myHeight, isCenter, formfie
 	return false;
 }
 
-function ipchoosewin(formfield, netid, device)
+function ipchoosewin(formfield1, formfield2, netid, device)
 {
-    var url = '?m=chooseip' +  (netid ? '&netid=' + netid : '') + (device ? '&device=' + device : '');
-	return openSelectWindow(url,'chooseip',350,380,'true',formfield);
+	var url = '?m=chooseip' +  (netid ? '&netid=' + netid : '') + (device ? '&device=' + device : '');
+	return openSelectWindow2(url, 'chooseip', 350, 380, 'true', formfield1, formfield2);
 }
 
 function macchoosewin(formfield)
diff --git a/lib/LMS.class.php b/lib/LMS.class.php
index a46c5f8..011aded 100644
--- a/lib/LMS.class.php
+++ b/lib/LMS.class.php
@@ -1479,6 +1479,7 @@ class LMS {
 			'nas' => isset($nodedata['nas']) ? $nodedata['nas'] : 0,
 			'longitude' => !empty($nodedata['longitude']) ? str_replace(',', '.', $nodedata['longitude']) : null,
 			'latitude' => !empty($nodedata['latitude']) ? str_replace(',', '.', $nodedata['latitude']) : null,
+			$SYSLOG_RESOURCE_KEYS[SYSLOG_RES_NETWORK] => $nodedata['netid'],
 			$SYSLOG_RESOURCE_KEYS[SYSLOG_RES_NODE] => $nodedata['id']
 		);
 		$this->DB->Execute('UPDATE nodes SET name=UPPER(?), ipaddr_pub=inet_aton(?),
@@ -1486,13 +1487,14 @@ class LMS {
 				modid=?, access=?, warning=?, ownerid=?, info=?, location=?,
 				location_city=?, location_street=?, location_house=?, location_flat=?,
 				chkmac=?, halfduplex=?, linktype=?, linkspeed=?, port=?, nas=?,
-				longitude=?, latitude=? 
+				longitude=?, latitude=?, netid=?
 				WHERE id=?', array_values($args));
 
 		if ($this->SYSLOG) {
 			unset($args[$SYSLOG_RESOURCE_KEYS[SYSLOG_RES_USER]]);
 			$this->SYSLOG->AddMessage(SYSLOG_RES_NODE, SYSLOG_OPER_UPDATE, $args,
-				array($SYSLOG_RESOURCE_KEYS[SYSLOG_RES_NODE], $SYSLOG_RESOURCE_KEYS[SYSLOG_RES_CUST]));
+				array($SYSLOG_RESOURCE_KEYS[SYSLOG_RES_NODE], $SYSLOG_RESOURCE_KEYS[SYSLOG_RES_NETWORK],
+					$SYSLOG_RESOURCE_KEYS[SYSLOG_RES_CUST]));
 
 			$macs = $this->DB->GetAll('SELECT id, nodeid FROM macs WHERE nodeid = ?', array($nodedata['id']));
 			if (!empty($macs))
@@ -1629,10 +1631,9 @@ class LMS {
 				$result['macs'][] = array('mac' => $mac, 'producer' => get_producer($mac));
 			unset($result['mac']);
 
-			if ($net = $this->DB->GetRow('SELECT id, name FROM networks
-				WHERE address = (inet_aton(?) & inet_aton(mask))', array($result['ip']))) {
-				$result['netid'] = $net['id'];
-				$result['netname'] = $net['name'];
+			if ($netname = $this->DB->GetOne('SELECT name FROM networks
+				WHERE id = ?', array($result['netid']))) {
+				$result['netname'] = $netname;
 			}
 
 			return $result;
@@ -1921,15 +1922,17 @@ class LMS {
 			'halfduplex' => $nodedata['halfduplex'],
 			'nas' => isset($nodedata['nas']) ? $nodedata['nas'] : 0,
 			'longitude' => !empty($nodedata['longitude']) ? str_replace(',', '.', $nodedata['longitude']) : null,
-			'latitude' => !empty($nodedata['latitude']) ? str_replace(',', '.', $nodedata['latitude']) : null
+			'latitude' => !empty($nodedata['latitude']) ? str_replace(',', '.', $nodedata['latitude']) : null,
+			$SYSLOG_RESOURCE_KEYS[SYSLOG_RES_NETWORK] => $nodedata['netid'],
 		);
 
 		if ($this->DB->Execute('INSERT INTO nodes (name, ipaddr, ipaddr_pub, ownerid,
 			passwd, creatorid, creationdate, access, warning, info, netdev,
 			location, location_city, location_street, location_house, location_flat,
-			linktype, linkspeed, port, chkmac, halfduplex, nas, longitude, latitude)
+			linktype, linkspeed, port, chkmac, halfduplex, nas, longitude, latitude,
+			netid)
 			VALUES (?, inet_aton(?), inet_aton(?), ?, ?, ?,
-			?NOW?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)', array_values($args))) {
+			?NOW?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)', array_values($args))) {
 			$id = $this->DB->GetLastInsertID('nodes');
 
 			// EtherWerX support (devices have some limits)
@@ -1954,7 +1957,8 @@ class LMS {
 				unset($args[$SYSLOG_RESOURCE_KEYS[SYSLOG_RES_USER]]);
 				$args[$SYSLOG_RESOURCE_KEYS[SYSLOG_RES_NODE]] = $id;
 				$this->SYSLOG->AddMessage(SYSLOG_RES_NODE, SYSLOG_OPER_ADD, $args,
-					array($SYSLOG_RESOURCE_KEYS[SYSLOG_RES_NODE], $SYSLOG_RESOURCE_KEYS[SYSLOG_RES_CUST]));
+					array($SYSLOG_RESOURCE_KEYS[SYSLOG_RES_NODE], $SYSLOG_RESOURCE_KEYS[SYSLOG_RES_NETWORK],
+						$SYSLOG_RESOURCE_KEYS[SYSLOG_RES_CUST]));
 			}
 
 			foreach ($nodedata['macs'] as $mac)
@@ -3539,8 +3543,11 @@ class LMS {
 			return $this->DB->Execute('UPDATE networks SET disabled = 1 WHERE id = ?', array($id));
 	}
 
-	function IsIPFree($ip) {
-		return !($this->DB->GetOne('SELECT id FROM nodes WHERE ipaddr=inet_aton(?) OR ipaddr_pub=inet_aton(?)', array($ip, $ip)) ? TRUE : FALSE);
+	function IsIPFree($ip, $netid = 0) {
+		if ($netid)
+			return !($this->DB->GetOne('SELECT id FROM nodes WHERE (ipaddr=inet_aton(?) AND netid=?) OR ipaddr_pub=inet_aton(?)', array($ip, $netid, $ip)) ? TRUE : FALSE);
+		else
+			return !($this->DB->GetOne('SELECT id FROM nodes WHERE ipaddr=inet_aton(?) OR ipaddr_pub=inet_aton(?)', array($ip, $ip)) ? TRUE : FALSE);
 	}
 
 	function IsIPGateway($ip) {
diff --git a/lib/SYSLOG.class.php b/lib/SYSLOG.class.php
index 054bf40..f7e1a2c 100644
--- a/lib/SYSLOG.class.php
+++ b/lib/SYSLOG.class.php
@@ -42,6 +42,7 @@ define('SYSLOG_RES_DNOTECONT', 39);
 define('SYSLOG_RES_CASHREG', 40);
 define('SYSLOG_RES_CASHRIGHT', 41);
 define('SYSLOG_RES_CASHREGHIST', 42);
+define('SYSLOG_RES_NETWORK', 43);
 
 $SYSLOG_RESOURCES = array(
 	SYSLOG_RES_USER => trans('user<!syslog>'),
@@ -86,6 +87,7 @@ $SYSLOG_RESOURCES = array(
 	SYSLOG_RES_CASHREG => trans('cash registry<!syslog>'),
 	SYSLOG_RES_CASHRIGHT => trans('cash registry rights<!syslog>'),
 	SYSLOG_RES_CASHREGHIST => trans('cash registry history<!syslog>'),
+	SYSLOG_RES_NETWORK => trans('network<!syslog>'),
 );
 
 $SYSLOG_RESOURCE_KEYS = array(
@@ -131,6 +133,7 @@ $SYSLOG_RESOURCE_KEYS = array(
 	SYSLOG_RES_CASHREG => 'cashregistryid',
 	SYSLOG_RES_CASHRIGHT => 'cashrightid',
 	SYSLOG_RES_CASHREGHIST => 'cashreghistoryid',
+	SYSLOG_RES_NETWORK => 'networkid',
 );
 
 define('SYSLOG_OPER_ADD', 1);
diff --git a/lib/locale/pl/strings.php b/lib/locale/pl/strings.php
index 2219b17..a8f8eaf 100644
--- a/lib/locale/pl/strings.php
+++ b/lib/locale/pl/strings.php
@@ -2717,6 +2717,7 @@ $_LANG['debit note contents<!syslog>'] = 'zawartość noty obciążeniowej';
 $_LANG['cash registry<!syslog>'] = 'rejestr kasowy';
 $_LANG['cash registry rights<!syslog>'] = 'uprawnienia do rejestru kasowego';
 $_LANG['cash registry history<!syslog>'] = 'historia rejestru kasowego';
+$_LANG['network<!syslog>'] = 'sieć';
 
 $_LANG['addition<!syslog>'] = 'dodanie';
 $_LANG['deletion<!syslog>'] = 'usunięcie';
@@ -2728,4 +2729,7 @@ $_LANG['log in failed<!syslog>'] = 'logowanie niepowodzeniem';
 $_LANG['log in<!syslog>'] = 'zalogowanie';
 $_LANG['log out<!syslog>'] = 'wylogowanie';
 
+$_LANG['Choose network'] = 'Wybierz sieć';
+$_LANG['- automatic selection -'] = '- wybór automatyczny -';
+
 ?>
diff --git a/modules/chooseip.php b/modules/chooseip.php
index 82266c1..85611db 100644
--- a/modules/chooseip.php
+++ b/modules/chooseip.php
@@ -31,7 +31,7 @@ $networks = $LMS->GetNetworks(true);
 $p = isset($_GET['p']) ? $_GET['p'] : '';
 
 if(!$p || $p == 'main')
-	$js = 'var targetfield = window.parent.targetfield;';
+	$js = 'var targetfield1 = window.parent.targetfield1;var targetfield2 = window.parent.targetfield2;';
 else
 	$js = '';
 
diff --git a/modules/netdevedit.php b/modules/netdevedit.php
index da606f9..d199c48 100644
--- a/modules/netdevedit.php
+++ b/modules/netdevedit.php
@@ -243,6 +243,7 @@ switch ($action) {
 		$subtitle = trans('New IP address');
 		$nodeipdata['access'] = 1;
 		$nodeipdata['macs'] = array(0 => '');
+		$SMARTY->assign('networks', $LMS->GetNetworks(true));
 		$SMARTY->assign('nodeipdata', $nodeipdata);
 		$edit = 'addip';
 		break;
@@ -256,6 +257,7 @@ switch ($action) {
 		foreach ($nodeipdata['macs'] as $key => $value)
 			$macs[] = $nodeipdata['macs'][$key]['mac'];
 		$nodeipdata['macs'] = $macs;
+		$SMARTY->assign('networks', $LMS->GetNetworks(true));
 		$SMARTY->assign('nodeipdata', $nodeipdata);
 		$edit = 'ip';
 		break;
@@ -310,8 +312,13 @@ switch ($action) {
 			$error['ipaddr'] = trans('Incorrect IP address!');
 		elseif (!$LMS->IsIPValid($nodeipdata['ipaddr']))
 			$error['ipaddr'] = trans('Specified address does not belongs to any network!');
-		elseif (!$LMS->IsIPFree($nodeipdata['ipaddr']))
-			$error['ipaddr'] = trans('Specified IP address is in use!');
+		else {
+			if (empty($nodeipadata['netid']))
+				$nodeipdata['netid'] = $DB->GetOne('SELECT id FROM networks WHERE INET_ATON(?) & mask = address ORDER BY id LIMIT 1',
+					array($nodeipdata['ipaddr']));
+			if (!$LMS->IsIPFree($nodeipdata['ipaddr'], $nodeipdata['netid']))
+				$error['ipaddr'] = trans('Specified IP address is in use!');
+		}
 
 		if ($nodeipdata['ipaddr_pub'] != '0.0.0.0' && $nodeipdata['ipaddr_pub'] != '') {
 			if (!check_ip($nodeipdata['ipaddr_pub']))
@@ -393,9 +400,14 @@ switch ($action) {
 			$error['ipaddr'] = trans('Incorrect IP address!');
 		elseif (!$LMS->IsIPValid($nodeipdata['ipaddr']))
 			$error['ipaddr'] = trans('Specified address does not belongs to any network!');
-		elseif (!$LMS->IsIPFree($nodeipdata['ipaddr']) &&
+		else {
+			if (empty($nodeipadata['netid']))
+				$nodeipdata['netid'] = $DB->GetOne('SELECT id FROM networks WHERE INET_ATON(?) & mask = address ORDER BY id LIMIT 1',
+					array($nodeipdata['ipaddr']));
+			if (!$LMS->IsIPFree($nodeipdata['ipaddr'], $nodeipdata['netid']) &&
 				$LMS->GetNodeIPByID($_GET['ip']) != $nodeipdata['ipaddr'])
-			$error['ipaddr'] = trans('IP address is in use!');
+				$error['ipaddr'] = trans('IP address is in use!');
+		}
 
 		if ($nodeipdata['ipaddr_pub'] != '0.0.0.0' && $nodeipdata['ipaddr_pub'] != '') {
 			if (check_ip($nodeipdata['ipaddr_pub'])) {
diff --git a/modules/nodeadd.php b/modules/nodeadd.php
index 30be318..ae444a3 100644
--- a/modules/nodeadd.php
+++ b/modules/nodeadd.php
@@ -52,6 +52,7 @@ if (isset($_POST['nodedata']))
 {
 	$nodedata = $_POST['nodedata'];
 
+	$nodedata['netid'] = $_POST['nodedatanetid'];
 	$nodedata['ipaddr'] = $_POST['nodedataipaddr'];
 	$nodedata['ipaddr_pub'] = $_POST['nodedataipaddr_pub'];
 	foreach($nodedata['macs'] as $key => $value)
@@ -84,10 +85,15 @@ if (isset($_POST['nodedata']))
 		$error['ipaddr'] = trans('Incorrect node IP address!');
 	elseif(!$LMS->IsIPValid($nodedata['ipaddr']))
 		$error['ipaddr'] = trans('Specified IP address doesn\'t overlap with any network!');
-	elseif(!$LMS->IsIPFree($nodedata['ipaddr']))
-		$error['ipaddr'] = trans('Specified IP address is in use!');
-	elseif($LMS->IsIPGateway($nodedata['ipaddr']))
-		$error['ipaddr'] = trans('Specified IP address is network gateway!');
+	else {
+		if (empty($nodedata['netid']))
+			$nodeedit['netid'] = $DB->GetOne('SELECT id FROM networks WHERE INET_ATON(?) & mask = address ORDER BY id LIMIT 1',
+				array($nodedata['ipaddr']));
+		if (!$LMS->IsIPFree($nodedata['ipaddr']))
+			$error['ipaddr'] = trans('Specified IP address is in use!');
+		elseif($LMS->IsIPGateway($nodedata['ipaddr']))
+			$error['ipaddr'] = trans('Specified IP address is network gateway!');
+	}
 
 	if($nodedata['ipaddr_pub']!='0.0.0.0' && $nodedata['ipaddr_pub']!='')
 	{
@@ -223,6 +229,7 @@ if(!isset($CONFIG['phpui']['big_networks']) || !chkconfig($CONFIG['phpui']['big_
 
 $nodedata = $LMS->ExecHook('node_add_init', $nodedata);
 
+$SMARTY->assign('networks', $LMS->GetNetworks(true));
 $SMARTY->assign('netdevices', $LMS->GetNetDevNames());
 $SMARTY->assign('error', $error);
 $SMARTY->assign('nodedata', $nodedata);
diff --git a/modules/nodeedit.php b/modules/nodeedit.php
index 2257f10..a82b22f 100644
--- a/modules/nodeedit.php
+++ b/modules/nodeedit.php
@@ -91,6 +91,7 @@ $layout['pagetitle'] = trans('Node Edit: $a', $nodeinfo['name']);
 if (isset($_POST['nodeedit'])) {
 	$nodeedit = $_POST['nodeedit'];
 
+	$nodeedit['netid'] = $_POST['nodeeditnetid'];
 	$nodeedit['ipaddr'] = $_POST['nodeeditipaddr'];
 	$nodeedit['ipaddr_pub'] = $_POST['nodeeditipaddrpub'];
 	foreach ($nodeedit['macs'] as $key => $value)
@@ -106,8 +107,11 @@ if (isset($_POST['nodeedit'])) {
 
 	if (check_ip($nodeedit['ipaddr'])) {
 		if ($LMS->IsIPValid($nodeedit['ipaddr'])) {
+			if (empty($nodeedit['netid']))
+				$nodeedit['netid'] = $DB->GetOne('SELECT id FROM networks WHERE INET_ATON(?) & mask = address ORDER BY id LIMIT 1',
+					array($nodeedit['ipaddr']));
 			$ip = $LMS->GetNodeIPByID($nodeedit['id']);
-			if ($ip != $nodeedit['ipaddr'] && !$LMS->IsIPFree($nodeedit['ipaddr']))
+			if ($ip != $nodeedit['ipaddr'] && !$LMS->IsIPFree($nodeedit['ipaddr'], $nodeedit['netid']))
 				$error['ipaddr'] = trans('Specified IP address is in use!');
 			elseif ($ip != $nodeedit['ipaddr'] && $LMS->IsIPGateway($nodeedit['ipaddr']))
 				$error['ipaddr'] = trans('Specified IP address is network gateway!');
@@ -267,6 +271,7 @@ include(MODULES_DIR . '/nodexajax.inc.php');
 $SMARTY->assign('xajax', $LMS->RunXajax());
 
 $SMARTY->assign('nodesessions', $LMS->GetNodeSessions($nodeid));
+$SMARTY->assign('networks', $LMS->GetNetworks(true));
 $SMARTY->assign('netdevices', $LMS->GetNetDevNames());
 $SMARTY->assign('nodegroups', $LMS->GetNodeGroupNamesByNode($nodeid));
 $SMARTY->assign('othernodegroups', $LMS->GetNodeGroupNamesWithoutNode($nodeid));
diff --git a/templates/default/chooseip.html b/templates/default/chooseip.html
index 5b9b821..d3f88e8 100644
--- a/templates/default/chooseip.html
+++ b/templates/default/chooseip.html
@@ -20,18 +20,29 @@
 {/if}
 {if $part eq "main"}
 <body>
+	<script type="text/javascript">
+	<!--
+	function send_address_settings(ipaddr) {
+		sendvalue(targetfield1, ipaddr);
+		if (targetfield2 != null)
+			for (var i = 0; i < targetfield2.options.length; i++)
+				if (targetfield2.options[i].value == '{$netid}')
+					targetfield2.options[i].selected = true;
+	}
+	//-->
+	</script>
 	<table width="100%" cellpadding="1">
 	{section name=iplist loop=$network.nodes.address}
 	{if $smarty.section.iplist.index is even}<TR>{/if}
-		<TD width="50%" align="center" class="highlight" {if $network.nodes.name[iplist] eq ""} onClick="sendvalue(targetfield,'{$network.nodes.address[iplist]}');"{/if}>
+		<TD width="50%" align="center" class="highlight" {if $network.nodes.name[iplist] eq ""} onClick="send_address_settings('{$network.nodes.address[iplist]}');"{/if}>
 			{if $network.nodes.name[iplist] neq ""}
 				{if $device && $network.nodes.name[iplist] eq "<b>GATEWAY</b>"}
-					<A href="javascript: void(0);" onClick="sendvalue(targetfield,'{$network.nodes.address[iplist]}');">{$network.nodes.name[iplist]}</A>
+					<A href="javascript: void(0);" onClick="send_address_settings('{$network.nodes.address[iplist]}');">{$network.nodes.name[iplist]}</A>
 				{else}
 					<FONT class="blend">{$network.nodes.name[iplist]}</FONT>
 				{/if}
 			{else}
-			<A href="javascript: void(0);" onClick="sendvalue(targetfield,'{$network.nodes.address[iplist]}');">{$network.nodes.address[iplist]}</A>
+			<A href="javascript: void(0);" onClick="send_address_settings('{$network.nodes.address[iplist]}');">{$network.nodes.address[iplist]}</A>
 			{/if}
 		</TD>
 	{if $smarty.section.iplist.index is not even}</TR>{/if}
diff --git a/templates/default/netdevipaddbox.html b/templates/default/netdevipaddbox.html
index ad6e90b..77d2fba 100644
--- a/templates/default/netdevipaddbox.html
+++ b/templates/default/netdevipaddbox.html
@@ -24,6 +24,22 @@
 					<TD WIDTH="50%" style="vertical-align:top;">
 						<TABLE WIDTH="100%" CELLPADDING="3">
 	<TR>
+		<TD WIDTH="1"%">
+			<IMG SRC="img/network.gif" ALT="">
+		</TD>
+		<TD WIDTH="1%" NOWRAP>
+			<B>{trans("Network:")}</B>
+		</TD>
+		<TD WIDTH="98%">
+			<SELECT NAME="ipadd[netid]" {tip text="Choose network" trigger="netid"}>
+				<OPTION value="0">{trans("- automatic selection -")}</OPTION>
+				{foreach $networks as $net}
+				<OPTION VALUE="{$net.id}" {if $net.id eq $nodeipdata.netid} selected{/if}{if $net.disabled eq 1} class="alert"{/if}>{$net.name|truncate:20:"":true}: {$net.address}/{$net.prefix}</OPTION>
+				{/foreach}
+			</SELECT>
+		</TD>
+	</TR>
+	<TR>
 		<TD WIDTH="1%">
 			<IMG SRC="img/ip.gif" ALT="">
 		</TD>
@@ -31,7 +47,7 @@
 			<B>{trans("IP address:")}</B>
 		</TD>
 		<TD WIDTH="98%">
-			<INPUT TYPE="TEXT" NAME="ipadd[ipaddr]" VALUE="{$nodeipdata.ipaddr}" {tip text="Enter IP address" trigger="ipaddr"}> <a href="javascript: void(0);" onClick="return ipchoosewin(document.editipaddr['ipadd[ipaddr]'],0,{$netdevinfo.id});" {tip text="Click to select IP from the list"}>&raquo;&raquo;&raquo;</A>
+			<INPUT TYPE="TEXT" NAME="ipadd[ipaddr]" VALUE="{$nodeipdata.ipaddr}" {tip text="Enter IP address" trigger="ipaddr"}> <a href="javascript: void(0);" onClick="return ipchoosewin(document.editipaddr['ipadd[ipaddr]'], document.editipaddr['ipadd[netid]'], 0, {$netdevinfo.id});" {tip text="Click to select IP from the list"}>&raquo;&raquo;&raquo;</A>
 		</TD>
 	</TR>
 	{if check_conf('phpui.public_ip')}
@@ -43,7 +59,7 @@
 			<B>{trans("Pub. IP address:")}</B>
 		</TD>
 		<TD WIDTH="98%">
-		        <INPUT TYPE="TEXT" NAME="ipadd[ipaddr_pub]" VALUE="{$nodeipdata.ipaddr_pub}" {tip text="Enter IP address (optional)" trigger="ipaddr_pub"}> <a href="javascript: void(0);" onClick="return ipchoosewin(document.editipaddr['ipadd[ipaddr_pub]'],0,{$netdevinfo.id});" {tip text="Click to select IP from the list"}>&raquo;&raquo;&raquo;</A>
+		        <INPUT TYPE="TEXT" NAME="ipadd[ipaddr_pub]" VALUE="{$nodeipdata.ipaddr_pub}" {tip text="Enter IP address (optional)" trigger="ipaddr_pub"}> <a href="javascript: void(0);" onClick="return ipchoosewin(document.editipaddr['ipadd[ipaddr_pub]'], null, 0, {$netdevinfo.id});" {tip text="Click to select IP from the list"}>&raquo;&raquo;&raquo;</A>
 		</TD>
 	</TR>
 	{/if}
diff --git a/templates/default/netdevipeditbox.html b/templates/default/netdevipeditbox.html
index e6951fd..57e5c8a 100644
--- a/templates/default/netdevipeditbox.html
+++ b/templates/default/netdevipeditbox.html
@@ -30,6 +30,22 @@
 					<TD WIDTH="50%" style="vertical-align:top;">
 						<TABLE WIDTH="100%" CELLPADDING="3">
 	<TR>
+		<TD WIDTH="1"%">
+			<IMG SRC="img/network.gif" ALT="">
+		</TD>
+		<TD WIDTH="1%" NOWRAP>
+			<B>{trans("Network:")}</B>
+		</TD>
+		<TD WIDTH="98%">
+			<SELECT NAME="ipadd[netid]" {tip text="Choose network" trigger="netid"}>
+				<OPTION value="0">{trans("- automatic selection -")}</OPTION>
+				{foreach $networks as $net}
+				<OPTION VALUE="{$net.id}" {if $net.id eq $nodeipdata.netid} selected{/if}{if $net.disabled eq 1} class="alert"{/if}>{$net.name|truncate:20:"":true}: {$net.address}/{$net.prefix}</OPTION>
+				{/foreach}
+			</SELECT>
+		</TD>
+	</TR>
+	<TR>
 		<TD WIDTH="1%">
 			<IMG SRC="img/ip.gif" ALT="">
 		</TD>
@@ -37,7 +53,7 @@
 			<B>{trans("IP address:")}</B>
 		</TD>
 		<TD WIDTH="98%">
-                       <INPUT TYPE="TEXT" NAME="ipadd[ipaddr]" VALUE="{$nodeipdata.ipaddr}" {tip text="Enter IP address" trigger="ipaddr"}> <a href="javascript: void(0);" onClick="return ipchoosewin(document.editipaddr['ipadd[ipaddr]'],{$nodeipdata.netid},{$netdevinfo.id});" {tip text="Click to select IP from the list"}>&raquo;&raquo;&raquo;</A>
+                       <INPUT TYPE="TEXT" NAME="ipadd[ipaddr]" VALUE="{$nodeipdata.ipaddr}" {tip text="Enter IP address" trigger="ipaddr"}> <a href="javascript: void(0);" onClick="return ipchoosewin(document.editipaddr['ipadd[ipaddr]'], document.editipaddr['ipadd[netid]'], {$nodeipdata.netid}, {$netdevinfo.id});" {tip text="Click to select IP from the list"}>&raquo;&raquo;&raquo;</A>
 		</TD>
 	</TR>
 	{if check_conf('phpui.public_ip')}
@@ -49,7 +65,7 @@
 			<B>{trans("Pub. IP address:")}</B>
 		</TD>
 		<TD WIDTH="98%">
-                       <INPUT TYPE="TEXT" NAME="ipadd[ipaddr_pub]" VALUE="{if $nodeipdata.ip_pub!="0.0.0.0"}{$nodeipdata.ip_pub}{/if}" {tip text="Enter IP address" trigger="ipaddr_pub"}> <a href="javascript: void(0);" onClick="return ipchoosewin(document.editipaddr['ipadd[ipaddr_pub]'],{$nodeipdata.netid},{$netdevinfo.id});" {tip text="Click to select IP from the list"}>&raquo;&raquo;&raquo;</A>
+                       <INPUT TYPE="TEXT" NAME="ipadd[ipaddr_pub]" VALUE="{if $nodeipdata.ip_pub!="0.0.0.0"}{$nodeipdata.ip_pub}{/if}" {tip text="Enter IP address" trigger="ipaddr_pub"}> <a href="javascript: void(0);" onClick="return ipchoosewin(document.editipaddr['ipadd[ipaddr_pub]'], null, {$nodeipdata.netid}, {$netdevinfo.id});" {tip text="Click to select IP from the list"}>&raquo;&raquo;&raquo;</A>
 		</TD>
 	</TR>
 	{/if}
diff --git a/templates/default/nodeaddbox.html b/templates/default/nodeaddbox.html
index 656c5df..855d129 100644
--- a/templates/default/nodeaddbox.html
+++ b/templates/default/nodeaddbox.html
@@ -21,10 +21,23 @@
 						<TABLE style="width: 100%;" CELLPADDING="3">
 	<TR>
 		<TD style="width: 1%;">
+			<IMG SRC="img/network.gif" ALT="{trans("Network:")}">
+		</TD>
+		<TD style="width: 99%;">
+			<SELECT SIZE="1" name="nodedatanetid" {tip text="Choose network" trigger="netid"}>
+				<OPTION VALUE="0">{trans("- automatic selection -")}</OPTION>
+				{foreach $networks as $net}
+				<OPTION VALUE="{$net.id}" {if $net.id eq $nodedata.netid} selected{/if}{if $net.disabled eq 1} class="alert"{/if}>{$net.name|truncate:20:"":true}: {$net.address}/{$net.prefix}</OPTION>
+				{/foreach}
+			</SELECT>
+		</TD>
+	</TR>
+	<TR>
+		<TD style="width: 1%;">
 			<IMG SRC="img/ip.gif" ALT="{trans("IP address:")}">
 		</TD>
 		<TD style="width: 99%;">
-			<INPUT TYPE="TEXT" NAME="nodedataipaddr" VALUE="{$nodedata.ipaddr}" {tip text="Enter IP address" trigger="ipaddr"}> <a href="javascript: void(0);" onClick="return ipchoosewin(document.nodeadd.nodedataipaddr);" {tip text="Click to select IP from the list"}>&raquo;&raquo;&raquo;</A>
+			<INPUT TYPE="TEXT" NAME="nodedataipaddr" VALUE="{$nodedata.ipaddr}" {tip text="Enter IP address" trigger="ipaddr"}> <a href="javascript: void(0);" onClick="return ipchoosewin(document.nodeadd.nodedataipaddr, document.nodeadd.nodedatanetid);" {tip text="Click to select IP from the list"}>&raquo;&raquo;&raquo;</A>
 		</TD>
 	</TR>
 	{if check_conf('phpui.public_ip')}
@@ -33,7 +46,7 @@
 			<IMG SRC="img/ip_pub.gif" ALT="{trans("Pub. IP address:")}">
 		</TD>
 		<TD style="width: 99%;">
-			<INPUT TYPE="TEXT" NAME="nodedataipaddr_pub" VALUE="{$nodedata.ipaddr_pub}" {tip text="Enter IP address (optional)" trigger="ipaddr_pub"}> <a href="javascript: void(0);" onClick="return ipchoosewin(document.nodeadd.nodedataipaddr_pub);" {tip text="Click to select IP from the list"}>&raquo;&raquo;&raquo;</A>
+			<INPUT TYPE="TEXT" NAME="nodedataipaddr_pub" VALUE="{$nodedata.ipaddr_pub}" {tip text="Enter IP address (optional)" trigger="ipaddr_pub"}> <a href="javascript: void(0);" onClick="return ipchoosewin(document.nodeadd.nodedataipaddr_pub, null);" {tip text="Click to select IP from the list"}>&raquo;&raquo;&raquo;</A>
 		</TD>
 	</TR>
 	{/if}
diff --git a/templates/default/nodeeditbox.html b/templates/default/nodeeditbox.html
index b778c30..837b7f7 100644
--- a/templates/default/nodeeditbox.html
+++ b/templates/default/nodeeditbox.html
@@ -23,10 +23,23 @@
 						<TABLE style="width: 100%;" CELLPADDING="3">
 	<TR>
 		<TD style="width: 1%;">
+			<IMG SRC="img/network.gif" ALT="{trans("Network:")}">
+		</TD>
+		<TD style="width: 99%;">
+			<SELECT SIZE="1" name="nodeeditnetid" {tip text="Choose network" trigger="netid"}>
+				<OPTION VALUE="0">{trans("- automatic selection -")}</OPTION>
+				{foreach $networks as $net}
+				<OPTION VALUE="{$net.id}" {if $net.id eq $nodeinfo.netid} selected{/if}{if $net.disabled eq 1} class="alert"{/if}>{$net.name|truncate:20:"":true}: {$net.address}/{$net.prefix}</OPTION>
+				{/foreach}
+			</SELECT>
+		</TD>
+	</TR>
+	<TR>
+		<TD style="width: 1%;">
 			<IMG SRC="img/ip.gif" ALT="{trans("IP address:")}">
 		</TD>
 		<TD style="width: 99%;" COLSPAN="2">
-			<INPUT TYPE="TEXT" NAME="nodeeditipaddr" VALUE="{$nodeinfo.ip}" {tip text="Enter IP address" trigger="ipaddr"}> <a href="javascript: void(0);" onClick="return ipchoosewin(document.editnode.nodeeditipaddr,{$nodeinfo.netid});" {tip text="Click to select IP from the list"}>&raquo;&raquo;&raquo;</A>
+			<INPUT TYPE="TEXT" NAME="nodeeditipaddr" VALUE="{$nodeinfo.ip}" {tip text="Enter IP address" trigger="ipaddr"}> <a href="javascript: void(0);" onClick="return ipchoosewin(document.editnode.nodeeditipaddr, document.editnode.nodeeditnetid, {$nodeinfo.netid});" {tip text="Click to select IP from the list"}>&raquo;&raquo;&raquo;</A>
 		</TD>
 	</TR>
 	<TR>
@@ -34,7 +47,7 @@
 			<IMG SRC="img/ip_pub.gif" ALT="{trans("Pub. IP address:")}">
 		</TD>
 		<TD style="width: 99%;" COLSPAN="2">
-			<INPUT TYPE="TEXT" NAME="nodeeditipaddrpub" {if $nodeinfo.ip_pub != "0.0.0.0"}VALUE="{$nodeinfo.ip_pub}"{/if} {tip text="Enter IP address (optional)" trigger="ipaddr_pub"}> <a href="javascript: void(0);" onClick="return ipchoosewin(document.editnode.nodeeditipaddrpub,{$nodeinfo.netid});" {tip text="Click to select IP from the list"}>&raquo;&raquo;&raquo;</A>
+			<INPUT TYPE="TEXT" NAME="nodeeditipaddrpub" {if $nodeinfo.ip_pub != "0.0.0.0"}VALUE="{$nodeinfo.ip_pub}"{/if} {tip text="Enter IP address (optional)" trigger="ipaddr_pub"}> <a href="javascript: void(0);" onClick="return ipchoosewin(document.editnode.nodeeditipaddrpub, null, {$nodeinfo.netid});" {tip text="Click to select IP from the list"}>&raquo;&raquo;&raquo;</A>
 		</TD>
 	</TR>
 	<TR>
diff --git a/templates/default/nodeinfobox.html b/templates/default/nodeinfobox.html
index fc86584..ca54b47 100644
--- a/templates/default/nodeinfobox.html
+++ b/templates/default/nodeinfobox.html
@@ -39,6 +39,14 @@
 						<TABLE style="width: 100%" CELLPADDING="3">
 	<TR>
 		<TD style="width: 1%;">
+			<IMG SRC="img/network.gif" ALT="{trans("Network:")}">
+		</TD>
+		<TD style="width: 99%;" COLSPAN="2">
+			{$nodeinfo.netname}
+		</TD>
+	</TR>
+	<TR>
+		<TD style="width: 1%;">
 			<IMG SRC="img/ip.gif" ALT="{trans("IP address:")}">
 		</TD>
 		<TD style="width: 99%;" COLSPAN="2">
