diff --git a/doc/lms.mysql b/doc/lms.mysql
index 72df5e583..a643d1424 100644
--- a/doc/lms.mysql
+++ b/doc/lms.mysql
@@ -840,12 +840,15 @@ CREATE TABLE rttickets (
   createtime int(11) NOT NULL DEFAULT '0',
   resolvetime int(11) NOT NULL DEFAULT '0',
   source tinyint(4) NOT NULL DEFAULT '0',
+  nodeid int(11) DEFAULT NULL,
   PRIMARY KEY (id),
   INDEX queueid (queueid),
   INDEX customerid (customerid),
   INDEX creatorid (creatorid),
   INDEX createtime (createtime),
-  FOREIGN KEY (queueid) REFERENCES rtqueues (id) ON DELETE CASCADE ON UPDATE CASCADE
+  FOREIGN KEY (queueid) REFERENCES rtqueues (id) ON DELETE CASCADE ON UPDATE CASCADE,
+  CONSTRAINT rttickets_nodeid_nodes_id_fk
+    FOREIGN KEY (nodeid) REFERENCES nodes (id) ON UPDATE CASCADE ON DELETE CASCADE
 ) ENGINE=InnoDB;
 
 # --------------------------------------------------------
@@ -2900,4 +2903,4 @@ INSERT INTO netdevicemodels (name, alternative_name, netdeviceproducerid) VALUES
 ('XR7', 'XR7 MINI PCI PCBA', 2),
 ('XR9', 'MINI PCI 600MW 900MHZ', 2);
 
-INSERT INTO dbinfo (keytype, keyvalue) VALUES ('dbversion', '2017070400');
+INSERT INTO dbinfo (keytype, keyvalue) VALUES ('dbversion', '2017070700');
diff --git a/doc/lms.pgsql b/doc/lms.pgsql
index 8b28ed3e7..eb5b7c583 100644
--- a/doc/lms.pgsql
+++ b/doc/lms.pgsql
@@ -1459,6 +1459,8 @@ CREATE TABLE rttickets (
   createtime integer 	DEFAULT 0 NOT NULL,
   resolvetime integer 	DEFAULT 0 NOT NULL,
   source smallint	DEFAULT 0 NOT NULL,
+  nodeid integer	DEFAULT NULL CONSTRAINT rttickets_nodeid_nodes_id_fk
+	FOREIGN KEY (nodeid) REFERENCES nodes (id) ON UPDATE CASCADE ON DELETE CASCADE,
   PRIMARY KEY (id)
 );
 
@@ -2964,6 +2966,6 @@ INSERT INTO netdevicemodels (name, alternative_name, netdeviceproducerid) VALUES
 ('XR7', 'XR7 MINI PCI PCBA', 2),
 ('XR9', 'MINI PCI 600MW 900MHZ', 2);
 
-INSERT INTO dbinfo (keytype, keyvalue) VALUES ('dbversion', '2017070400');
+INSERT INTO dbinfo (keytype, keyvalue) VALUES ('dbversion', '2017070700');
 
 COMMIT;
diff --git a/lib/LMSDB_common.class.php b/lib/LMSDB_common.class.php
index 5ef35b996..f3ef21256 100644
--- a/lib/LMSDB_common.class.php
+++ b/lib/LMSDB_common.class.php
@@ -24,7 +24,7 @@
  *  $Id$
  */
 
-define('DBVERSION', '2017070600'); // here should be always the newest version of database!
+define('DBVERSION', '2017070700'); // here should be always the newest version of database!
 				 // it placed here to avoid read disk every time when we call this file.
 
 /**
diff --git a/lib/upgradedb/mysql.2017070700.php b/lib/upgradedb/mysql.2017070700.php
new file mode 100644
index 000000000..93349f4a2
--- /dev/null
+++ b/lib/upgradedb/mysql.2017070700.php
@@ -0,0 +1,34 @@
+<?php
+
+/*
+ * LMS version 1.11-git
+ *
+ *  (C) Copyright 2001-2017 LMS Developers
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License Version 2 as
+ *  published by the Free Software Foundation.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307,
+ *  USA.
+ *
+ */
+
+$this->BeginTrans();
+
+$this->Execute("ALTER TABLE rttickets ADD COLUMN nodeid int(11) DEFAULT NULL");
+$this->Execute("ALTER TABLE rttickets ADD CONSTRAINT rttickets_nodeid_nodes_id_fk
+	FOREIGN KEY (nodeid) REFERENCES nodes (id) ON UPDATE CASCADE ON DELETE CASCADE");
+
+$this->Execute("UPDATE dbinfo SET keyvalue = ? WHERE keytype = ?", array('2017070700', 'dbversion'));
+
+$this->CommitTrans();
+
+?>
diff --git a/lib/upgradedb/postgres.2017070700.php b/lib/upgradedb/postgres.2017070700.php
new file mode 100644
index 000000000..1ad1b3fee
--- /dev/null
+++ b/lib/upgradedb/postgres.2017070700.php
@@ -0,0 +1,36 @@
+<?php
+
+/*
+ * LMS version 1.11-git
+ *
+ *  (C) Copyright 2001-2017 LMS Developers
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License Version 2 as
+ *  published by the Free Software Foundation.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307,
+ *  USA.
+ *
+ */
+
+$this->BeginTrans();
+
+$this->Execute("
+	ALTER TABLE rttickets ADD COLUMN nodeid integer DEFAULT NULL;
+	ALTER TABLE rttickets ADD CONSTRAINT rttickets_nodeid_nodes_id_fk
+		FOREIGN KEY (nodeid) REFERENCES nodes (id) ON UPDATE CASCADE ON DELETE CASCADE
+");
+
+$this->Execute("UPDATE dbinfo SET keyvalue = ? WHERE keytype = ?", array('2017070700', 'dbversion'));
+
+$this->CommitTrans();
+
+?>
