--- sysplugins/smarty_internal_resource_file.php	2015-07-04 15:02:11.885002326 +0200
+++ sysplugins/smarty_internal_resource_file.php	2015-07-04 13:39:53.030504828 +0200
@@ -53,7 +53,7 @@
             $_directories = $source->smarty->getTemplateDir();
         }
         // template_dir index?
-        if (!empty($fileMatch['index'])) {
+        if (isset($fileMatch['index'])) {
             $index = $fileMatch['index'];
             $_directory = null;
             // try string indexes
