--- sysplugins/smarty_internal_runtime_inheritance.php	2016-02-14 22:32:27.155339660 +0100
+++ sysplugins/smarty_internal_runtime_inheritance.php	2016-02-14 22:31:51.000000000 +0100
@@ -71,7 +71,7 @@
     public function init(Smarty_Internal_Template $tpl, $initChild, $blockNames = array())
     {
         // if template was from an inner block or template is a parent template create new inheritance root
-        if ($initChild && ($this->blockNesting || $this->state == 3)) {
+        if ($initChild && ($this->blockNesting || $this->state == 3) && strpos($tpl->template_resource, 'extendsall') === false) {
             $tpl->ext->_inheritance = new Smarty_Internal_Runtime_Inheritance();
             $tpl->ext->_inheritance->init($tpl, $initChild, $blockNames);
             return;
